### DISCUSSION TITLE
Rebalancing Firearm progression: An outline & Discussion

### DISCUSSION THREAD

**Discussion by ghost** [cataclysmbn/Cataclysm-BN#2693] [General]
It has gone around several times that progression sucks, I will outline my proposed ideas and what needs to happen FIRST in order to make this outline work in a fun and functional manner.

1. Keep volume of gun spawns as they are. NP's loot category system could be implemented in some form so you can just lower guns to nothing.

2. Implement the FBI regulations on military grade guns so less high tier guns spawn in civilian locations. Finding a working M249 SAW in a basement or an RPG at a drug deal is horrible for progression for numerous reasons even if ammunition was made hard to find for it.

3. Rework faults so that you can specify military grade hardware on combat zones needs repair by a mid level player, using common components and tools already available in game. Without repair, these military zombie guns aren't useful to you so they don't ruin early progression.

4. Add more casings around the world and make production of bullets and gunpowder easier, but lessen ammunition spawns on the logic that they were fired during the cataclysm.

**In order to make these 4 points happen, Here is what needs to be done FIRST.**

1. Again, implementing a loot category system so players can choose to lower or raise guns, lowering allows you to play more like NEO Scavenger as was suggested. Higher means a fast playstyle.

2. We need to research every gun in the game and if its hard to be owned as a civilian, redo how they spawn and rarities. Redo loot tables.

3. Faults need to be reworked in the code base to an acceptable state. The UI needs drastic improvement, it needs to be a fun mechanic or at least not an annoying one. Less keystrokes via perhaps a mend all option.

4. We need the ability for NPC's to craft for the player using a queue based crafting system and assigning work tasks. This should ideally include farming and possibly hunting, where they don't need to be told to keep farming and will do it year after year. As for manufacturing a future PR for getting factory machines back online would let you have more feasible crafting recipes and them getting done faster. We don't actually need to make the current handloading any easier if you were using a factory and a dozen npc's for the job.

Addendum 1: An expanded system for vehicles to be locked up and require opening from various methods. SWAT vehicles and military vehicles should be locked up tight and that would require the mechanics to hacksaw the door off if you can't open the lock.

[a comment by an automated account is not shown]

**Reply by ghost** [reply]
Having a working deathmobile is a way to massacre zombies in of itself so requiring mechanics to access a SWAT van is good, since you would at that point be building a functional car anyway for ramming zombies.

**Comment by DrPariah** [comment]
1) Bringing over the sliders from NP is a _terrific_ idea, and I'm all for it.
2) The rest of it...not so much, because what you're describing is a very specific meta that is the kind enforced by C:DDA and applied by mods like 'No Hope'. If there's opt out capability tweaking to taste so the individual can natively set their parameters from 'scavenge your brass and get your unpaid interns to handload for you so your machine shopped .50 Ersatz Rifle will have ammo because you couldn't get anything better thanks to real world FBI controls and civilian gun ownership statistics' to 'this is Gamma World by way of Cyberpunk 2077, **********ers, LOOT THE UNTOUCHED AMMO SUPPLIES LET'S GOOOOOOOOO' without having to go through the hassle of the 'realism' imposed by mainline DDA, I'd be down for that. But being forced into the 'unpaid intern bullet farm Ersatz Rifle' game would not be fun for me, and I doubt I'm the only one.

**Reply by ghost** [reply]
Balance is hard. I think having a mainline difficulty that is more difficult and allowing robust modding tools to make it easier AND harder is more useful for everyone. The faults system could include a way to disable faults altogether (fixing that part of the balancing) And the loot category system could potentially be programmed to let you change what specific items spawn in a itemgroup, without making a JSON mod solely to edit item groups. There is a lot of code specific solutions to make things infinitely more customizable than they are now, and also create a more engaging baseline.
